src/states/adult/mod.rs

```diff
@@ -493,31 +428,16 @@ impl Base for Adult {
 
         let mut transition = Transition::Stay;
 
-        for msg in mem::take(&mut self.msg_backlog) {
+        for QueuedMessage { sender, message } in
+            mem::replace(&mut self.msg_backlog, Default::default())
+        {
             if let Transition::Stay = &transition {
-                let result = match msg {
-                    QueuedMessage::Hop(msg) => {
-                        let msg = match HopMessageWithBytes::new(msg, &self.log_ident()) {
-                            Ok(msg) => msg,
-                            Err(err) => {
-                                error!("{} - Failed to make message {:?}", self, err);
-                                continue;
-                            }
-                        };
-
-                        self.handle_filtered_signed_message(msg)
-                    }
-                    QueuedMessage::Direct(sender, msg) => {
-                        self.handle_direct_message(msg, sender, outbox)
-                    }
-                };
-
-                match result {
+                match self.handle_message(sender, message, outbox) {
```

Comment:
This does not do the verify of the message that was done before this commit since it happened in `handle_filtered_signed_message`.